opal/mca/common/ofi/common_ofi.c

```diff
@@ -828,15 +852,10 @@ struct fi_info *opal_common_ofi_select_provider(struct fi_info *provider_list,
     /* Cycle through remaining fi_info objects, looking for alike providers */
     while (NULL != current_provider) {
         if (!check_provider_attr(provider, current_provider)) {
-            near = false;
 #if OPAL_OFI_PCI_DATA_AVAILABLE
-            if (NULL != current_provider->nic
-                && NULL != current_provider->nic->bus_attr
-                && current_provider->nic->bus_attr->bus_type == FI_BUS_PCI) {
-                pci = current_provider->nic->bus_attr->attr.pci;
-                near = is_near(distances, num_distances,
-                               opal_hwloc_topology, pci);
-            }
+            near = is_near(distances, num_distances, current_provider);
+#else
+            near = false;
```

Comment:
nit: I think it would be cleaner to define `is_near` regardless of OPAL_OFI_PCI_DATA_AVAILABLE's value.

Reply:
This has been addressed as part of the coverity fix PR. So there is no need to rework this fix here.

Reply:
> nit: I think this should be lower-case only.

I don't have a preference. Changed to case sensitive compare in the update.